pkg/gadgets/trace/dns/tracer/gadget.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2022-2023 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tracer
```

Comment:
We are missing this for gadgets that have a container column hidden by default:

https://github.com/inspektor-gadget/inspektor-gadget/blob/e348150b47821742f6e4f14d6ce56c8d7624e547/cmd/local-gadget/trace/dns.go#L52-L55

Resulting in:

```bash
$ sudo ./local-gadget trace dns
WARN[0000] Runtime enricher (cri-o): couldn't get current containers
QR NAMESERVER                  TYPE      QTYPE      NAME                                              RCODE
Q  10.0.2.3                    OUTGOING  A          www.example.com.
Q  10.0.2.3                    OUTGOING  AAAA       www.example.com.
R  10.0.2.3                    HOST      A          www.example.com.                                  NoError
R  10.0.2.3                    HOST      AAAA       www.example.com.                                  NoError
```
Maybe it can be done by the local runtime for all the gadgets that have such `container` column?

Reply:
Ah, good catch!

The local runtime is (despite its name) however not only used for the local-gadget part.

I suggest adding a tag to the container field and then wire it up [like we do with the columnfilters in the main.go of local-gadget/ig](https://github.com/inspektor-gadget/inspektor-gadget/pull/1281/files#diff-dd1c4cbfe354487800edab999f0c82690691718e40f00a208bba45ffa0dc07e4R47-R48).

Columns lib would need to get another helper function like `ShowColumnsWithTags([]string{})` that can be passed.

WDYT?

Reply:
> The local runtime is (despite its name) however not only used for the local-gadget part.
>
Oops, right.

> I suggest adding a tag to the container field.
>
It already has `kubernetes` and `runtime`. Why we are not using `runtime` for `local-gadget`? I think the parser should be initialized with the tag to use, then `GetDefaultColumns` should return no-tagged columns + the ones with the requested tag. WDYT?

> [like we do with the columnfilters in the main.go of local-gadget/ig](https://github.com/inspektor-gadget/inspektor-gadget/pull/1281/files#diff-dd1c4cbfe354487800edab999f0c82690691718e40f00a208bba45ffa0dc07e4R47-R48).
>
Link is not pointing me anywhere 🙁

Reply:
> > The local runtime is (despite its name) however not only used for the local-gadget part.
> 
> Oops, right.
> 
> > I suggest adding a tag to the container field.
> 
> It already has `kubernetes` and `runtime`. Why we are not using `runtime` for `local-gadget`? I think the parser should be initialized with the tag to use, then `GetDefaultColumns` should return no-tagged columns + the ones with the requested tag. WDYT?

It's already implemented that way (should be shown in the link if that'd work - `cmd/local-gadget/main.go` ll 47-48). The Problem is that `Container` gets manually hidden inside the `GetColumns()` of the gadgets' type package and we're not changing visibility based on the columns selected by tags, yet...

Reply:
No no, it is fine haha 🙂  ... Let me think about it